### DISCUSSION TITLE
Should areas be a subclass of zones?

### DISCUSSION THREAD

**Discussion by rlippmann** [home-assistant/architecture#1161] [Entity Models] [answered]
Hi all.

I was thinking about doing room presence in HA.  Preferably without having to buy sensors for that purpose.

I could make a zone for each room, but that seems tedious and redundant.

If areas subclassed zones, you can see if a location sensor was in a room.  Granted, you'd also have to set up each room's latitude/longitude/radius, but most cell phones have built in compasses with pretty accurate latitude/longitude data.  Maybe could even add that functionality to the companion apps.

You'd also need to deal with elevation, but since areas can be attached to floors, that could be doable by having a floor elevation.

Thoughts?

**Comment by MartinHjelmare** [comment]
It's not clear what you propose. A zone is an Entity. An area is not an Entity. Do you mean that areas should become Entity instances?

When opening an architecture issue you should have thought through the proposal and what is needed to realize the proposed design, and you should be willing to implement the changes. Please clarify your proposal and describe how the current architecture would need to change to accomplish the goals of the proposal, and what it would mean for the different users of the involved APIs.

**Reply by rlippmann** [reply]
I don't see why areas shouldn't be entities.  They're basically groups without the restriction of having to be the same entity type.

**Reply by rlippmann** [reply]
I'm proposing:

```
class Area(Zone):
   (whatever attributes Area currently has)
```

that way anything you can do with a zone (enter, leave, etc) can also be done with an area.

**Reply by MartinHjelmare** [reply]
To get any traction on this issue, I think you will need to explain a lot more how this would be achieved and what the consequences would be, including positive and negative aspects.

**Comment by tetele** [comment]
Although I do like the idea of being able to enter/exit an `area` and for an `area` entity to have a state reflecting the number of people/device trackers within, that's as far as this makes sense for me.

There's no way that GPS is accurate enough indoors (especially in multi-story buildings) to detect in which room you are. Plus, Zones are circular. Plus, depending on phone model, OS and settings, location may be determined using a myriad of sensors (e.g. connected WiFi, detected WiFi networks, mobile network triangularization, GPS etc.). Plus you seem to be treating *"that could be doable by having a floor elevation"* as if the elevation is a given :)

This seems like a very simplistic approach to an idea you had for fixing some urgent issue without understanding the implications.

**Comment by frenck** [comment, marked as answer]
Thanks for submitting an architectural proposal!

We discussed this proposal with the team during a core meeting and concluded that we would not move forward with it. Areas are designed to be parts inside your home, while zones are designed to be outside of it. We would like to keep this separate and not mix it up.

Nevertheless, thanks for participating in our design.

../Frenck
